smart-contracts/wasm-chain-integration/src/utils.rs

```diff
@@ -64,10 +65,14 @@ impl<I> machine::Host<I> for TrapHost {
 }
 
 /// A host which traps for any function call apart from `report_error` which it
-/// prints to standard out.
-pub struct TestHost;
+/// prints to standard out and `get_random` that calls a random number
+/// generator.
+pub struct TestHost<R: RngCore> {
+    /// A RNG for randomised testing.
+    rng: Option<R>,
```

Comment:
Why is this an Option? When will not be Some?

Reply:
Schema generation uses `TestHost`: https://github.com/Concordium/concordium-base/blob/f526e3e60f048a26f62cdd3073fa03ff995f2425/smart-contracts/wasm-chain-integration/src/utils.rs#L242

Reply:
Schema generation uses `TestHost` https://github.com/Concordium/concordium-base/blob/quickcheck/smart-contracts/wasm-chain-integration/src/utils.rs#L244

Reply:
Also, in `benches`. I've added a constructor that produces a `TestHost` instance with uninitialised RNG.